### PR TITLE
Add logic to review applicants

### PR DESCRIPTION
**Pull request by reefdog**
This PR adds the ability to review applicants (approve or reject) from the console using the `Applicant#approve` and `Applicant#revoke` methods. The applicant must have already confirmed their email address. Review notes are optional.

⚠️ This is just the first of several steps toward completion of #271, so it doesn't actually create a user yet, nor does it offer an interface in the site for this (coming #273).

**Testing:**
1. 🚨 `bundle install` for a necessary `sorbet-rails` update
2. 🚨 `rails db:migrate`
3. `rails t test/models/applicant_test.rb`
4. Start Rails and Sidekiq, create a new request for access, and confirm its email address
5. `rails c` and find your new applicant (`a = Applicant.last` perhaps)
6. Approve or reject with `a.approve` or `a.reject`
   - Optionally, provide a review note by passing `review_note:` or `review_note_internal:` to those methods
7. ✅ Verify approval/rejection with `a.approved?` and `a.rejected?` or the blanket `a.reviewed?` (true for both), remembering that you may need to `a.reload`

**Testing failure cases:**
- You should **not** be able to review applicants whose emails aren't confirmed (`confirmed_at == nil`).
- You should **not** be able to reject applicants that were already reviewed (`status != nil`).
- You should **not** be able to approve applicants that were already approved (`status == "approved"`).
   - Note the implication of this last one: you **should** be able to approve applicants that were previously rejected. This is intentional.

👋  The easiest way to reset an applicant is just to null out the `status` column manually rather than create a whole new applicant.

Affects #271

### CODE REVIEW

File: app/models/applicant.rb [hunk 1]
```diff
@@ -45,6 +50,65 @@ def self.find_unconfirmed_by_email_and_token(email:, token:)
...
+  def reject(
+    review_note: nil,
+    review_note_internal: nil
+  )
+    raise UnconfirmedError unless self.confirmed?
+    raise StatusChangeError if self.reviewed?
+
+    self.update!({
+      status: :rejected,
+      review_note:,
```
Comment (oneroyalace) [inline app/models/applicant.rb:96]: I believe @cguess prefers explicit key/value syntax for hashes

**Review by oneroyalace — APPROVED** [review]
LGTM. Side-note: I can't get emails to send anymore. Do you see any failures on the mailgun side?

File: app/models/applicant.rb (hunk 1 above)
Comment (reefdog) [reply app/models/applicant.rb:96]: I felt a little bit dirty myself, but I do love these implicit values. @cguess what do you think? (I'm _super_ used to it from JS, where the syntax is even cleaner and doesn't require the trailing colon, but if you want to keep it out of Ruby, I get it.)

Comment (reefdog) [reply app/models/applicant.rb:96]: @cguess [said](https://github.com/TechAndCheck/zenodotus/pull/340#discussion_r968572149):

> I'd prefer to keep it out. I just like explicitness.

Will change!

**Comment by reefdog** [comment]
Oh, for history's sake I should probably note that we determined the sending issue above was just Sidekiq not running locally. (I'd forgotten to note in the PR desc that it needed to be running. It always does, for mail delivery, as we use `deliver_later`.)

**Pull request merged**